service/handlers.go

```diff
@@ -197,11 +195,20 @@ func updateUserData(c *gin.Context) {
 
 	r := data_store.NewUserDater()
 	if r == nil {
-		c.JSON(http.StatusInternalServerError, INIT_USERDATA_FAIL)
+		c.JSON(http.StatusInternalServerError, outputErrorFormat(INIT_USERDATA_FAIL))
 	}
 
 	req := c.Request.RequestURI
 	method := c.Request.Method
 	activity := fmt.Sprintf("%s %s", method, req)
-	r.AddActivity(username, activity)
+	err := r.AddActivity(username, activity)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, outputErrorFormat("Failed To Update User Data"))
```

Comment:
you should return an error, and the caller of this function should handle the err